src/test/java/com/artipie/asto/StorageExclusivelyTest.java

```diff
@@ -81,6 +81,39 @@ void shouldRunExclusivelyWhenPrevFinished(final Storage storage) {
         );
     }
 
+    @TestTemplate
+    void shouldRunExclusivelyWhenPrevFinishedWithAsyncFailure(final Storage storage) {
+        final Key key = new Key.From("shouldRunExclusivelyWhenPrevFinishedWithAsyncFailure");
+        final FakeOperation operation = new FakeOperation();
```

Comment:
@olegmoz I'd suggest to inline `operation`, it's used only once

Reply:
@olenagerasimova the variable is used twice: first to complete it's `finished` future with exception (line 88) and then to pass it to `exclusively` method (line 91), so it cannot be inlined